### DISCUSSION TITLE
Azure AKS - Migration of PVCs from One Subscription to Another Subscription (Tenant is same for both subscriptions)

### DISCUSSION THREAD

**Discussion by zohebk8s** [kubernetes-sigs/azuredisk-csi-driver#1825] [Show and tell]
Use-Case: Kindly refer to the attachment for detailed information.

Azure AKS: Migrating PVCs from One Subscription to Another Subscription(In this POC/Demo, Tenant
is the same for both subscriptions)

High-Level Steps: (In this demo/POC, I've shown how we can migrate a PVC in between subscriptions)

1. Install MongoDB in Azure subscription 1---> (I've used MongoDB in this demo)
2. Create a Snapshot from the Azure Disk in Azure subscription 1
3. Move Snapshot from Azure subscription 1 to zoh-aks-poc
4. Create a Disk from the Snapshot of zoh-aks-poc
5. While creating a Disk, make sure to select zone. Otherwise it won't work. As the VM/PVC
should be in the same Zone.
6. Create a PV and PVC in zoh-aks-poc---> AKS (Use ZRS Storage Class)
7. Install MongoDB in zoh-aks-poc---> AKS
8. Exec into the pod and validate whether the data is available or not.
Important Note: Kindly test this use case in your sandbox/lower environment(Before
deploying in Production). If it’s working as expected, then promote it accordingly
[Azure AKS - PVC Migration between Subscriptions.pdf](https://github.com/kubernetes-sigs/azuredisk-csi-driver/files/11374069/Azure.AKS.-.PVC.Migration.between.Subscriptions.pdf)

**Comment by andyzhangx** [comment]
thanks for providing the cross subs migration solution, if it requires zero downtime, you should use disk snapshot or disk cloning (disk clone is a little convenient).

**Reply by zoheb-k8s** [reply]
@andyzhangx Thank you for acknowledging my efforts. In the attachment (My POC), I've followed the same approach. 
1. Create a snapshot from Azure Disk and move it to another subscription.
2. Create a disk from the migrated snapshot and then PV/PVC creation from that Disk.

Will you be able to add more details about disk cloning you mentioned?

**Reply by andyzhangx** [reply]
https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/cloning#use-volume-cloning-feature-to-create-a-copy-of-a-disk-with-a-different-sku

**Reply by zohebk8s** [reply]
@andyzhangx Thanks for sharing the details.

If we want to clone/create a PVC from an existing PVC, then the pre-requisite is - The existing PVCs should be available/deployed in that subscription.

**Migration of PVC from One Subscription to Another Subscription**: What I am trying to say is, in Subscription-A we have PVCs and Subscription B is completely empty and it doesn't have any Disks/PVCs. Now in order to clone/create a PVC from Subscription-A, we cannot use the "dataSource" option as we don't have any PVCs in Subscription B for reference. That's the reason I did the above POC/Demo, which highlights the migration/PVC creation part in other subscriptions. (Subscription B)

The document/URL(https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/deploy/example/cloning) which you have shared is really helpful - When we have PVCs already available in subscription and we want a copy of it.
